clippy_lints/src/absolute_symbol_paths.rs

```diff
@@ -0,0 +1,97 @@
+use clippy_utils::diagnostics::span_lint;
+use clippy_utils::source::snippet_opt;
+use rustc_data_structures::fx::FxHashSet;
+use rustc_hir::def::{DefKind, Res};
+use rustc_hir::def_id::{DefId, CRATE_DEF_INDEX};
+use rustc_hir::{HirId, ItemKind, Node, Path};
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_session::{declare_tool_lint, impl_lint_pass};
+use rustc_span::symbol::kw;
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for usage of symbols through absolute paths, like `std::env::current_dir`.
```

Comment:
I think we should drop the word `symbol` from the description/lint name, `definition` might be more accurate compiler wise but avoiding giving it a term might be best. Or maybe saying `item` even though it isn't 100% accurate